ui/src/pages/AllStories.tsx

```diff
@@ -0,0 +1,250 @@
+      prevState.length === rows.length ? [] : rows.map((d) => d.id)
+    );
+  };
+  const handleChecked = (e, d) => {
+    if (e.target.checked) {
+      setSelectedRowIds((prevState) => [...prevState, d.id]);
+    } else {
+      setSelectedRowIds((prevState) => prevState.filter((e) => e !== d.id));
+    }
+  };
+
+  const handleRequestSort = (property) => {
+    const isDesc = orderBy === property && order === "desc";
+    const newOrder = isDesc ? "asc" : "desc";
+    setOrder(newOrder);
+    setOrderBy(property);
+
+    onChangeTableSort(newOrder, property);
+  };
+  const onChangeTableSort = (order, property) => {
+    const newData = stableSort(tableData, getComparator(order, property));
+    setTableData(newData);
+  };
+  const stableSort = (array, comparator) => {
+    const stabilizedThis = array.map((el, index) => [el, index]);
+    stabilizedThis.sort((a, b) => {
+      const order = comparator(a[0], b[0]);
+      if (order !== 0) return order;
+      return a[1] - b[1];
+    });
+    return stabilizedThis.map((el) => el[0]);
+  };
+  const getComparator = (order, orderBy) => {
+    return order === "desc"
+      ? (a, b) => descendingComparator(a, b, orderBy)
+      : (a, b) => -descendingComparator(a, b, orderBy);
+  };
+  const descendingComparator = (a, b, orderBy) => {
+    if (b[orderBy] < a[orderBy]) {
+      return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+      return 1;
+    }
+    return 0;
+  };
+
+  const indeterminate =
+    selectedRowIds.length > 0 && selectedRowIds.length !== rows.length;
+
+  if (error) return <div>Error returning stories data!</div>;
+  if (!allStories) return <div>Loading all stories table..</div>;
+
+  const handleTabChange = (
+    event: React.ChangeEvent<Record<string, unknown>>,
+    newValue: number
+  ) => {
+    setTabValue(newValue);
+  };
+  return (
+    <>
+      <StyledAppBar position="relative">
+        <Tabs
+          value={tabValue}
+          onChange={handleTabChange}
+          aria-label="all stories tabs"
+        >
+          <Tab label="ALL STORIES" {...a11yProps(0)} />
+          <Tab label="VISIBLE CHANGES" {...a11yProps(1)} />
+          <Tab label="PENDING MAP CHANGES" {...a11yProps(2)} />
+        </Tabs>
+      </StyledAppBar>
+      <AllStoriesTabs value={tabValue} index={0}>
+        <VirtualizedTable
+          data={stableSort(tableData, getComparator(order, orderBy))}
+          order={order}
+          orderBy={orderBy}
+          columns={[
+            {
+              name: "id",
+              width: 100,
+              onHeaderClick() {
+                handleRequestSort("id");
+              },
+              header: (
+                <div>
+                  <Checkbox
+                    checked={selectedRowIds.length > 0}
+                    indeterminate={indeterminate}
+                    onChange={handleCheckedAll}
+                  />
+                  ID
+                </div>
+              ),
+              cell: (d) => (
```

Comment:
whats d here lol?

Reply:
you can actually just do `cell: ()` but it gives a lint warning for anonymous function so I just defined a param there.